qubes/devices.py

```diff
@@ -56,6 +55,34 @@ class DeviceAlreadyAttached(qubes.exc.QubesException, KeyError):
     '''Trying to attach already attached device'''
     pass
 
+class WrongAssignment(qubes.exc.QubesException, KeyError):
+    '''Trying to attach non permanent assignment to a halted vm'''
```

Comment:
I don't see a need for this exception any more. 2 of 3 use cases can be replaced by `qubes.exc.QubesVMNotHaltedError` & `qubes.exc.QubesVMNotRunningError` and the last one with an assert.

I will remove it